ipfs-3-to-4/migration/migration.go

```diff
@@ -387,6 +391,38 @@ func transferBlocks(flatfsdir string) error {
 
 	fmt.Println()
 
+	err := cleanEmptyDirs(flatfsdir)
+	if err != nil {
+		fmt.Println(err)
+	}
+
+	return nil
+}
+
+func cleanEmptyDirs(dir string) error {
+	children, err := ioutil.ReadDir(dir)
+	if err != nil {
+		return err
+	}
+
+	for _, c := range children {
+		if !c.IsDir() {
+			continue
+		}
+
+		cdir := filepath.Join(dir, c.Name())
+		blocks, err := ioutil.ReadDir(cdir)
```

Comment:
nit: File.Readdirnames would be a better choice.  ioutil.ReadDir will perform a stat on each directory entry and then sort them by name.  Not sure if it really matters so fell free to ignore.